File: src/main/java/com/example/spacemines/FlywayTestConfig.java
package com.example.spacemines;

import org.flywaydb.core.Flyway;
import org.springframework.boot.autoconfigure.flyway.FlywayMigrationStrategy;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class FlywayTestConfig {

    @Bean
    public FlywayMigrationStrategy cleanMigrateStrategy() {
        return (Flyway flyway) -> {
                flyway.clean();
                flyway.migrate();
        };
    }

}

File: src/test/kotlin/com/twcrone/spacemines/config/FlywayTestConfig.kt
package com.twcrone.spacemines.config

import org.springframework.boot.autoconfigure.flyway.FlywayMigrationStrategy
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class FlywayTestConfig {

    @Bean
    fun cleanMigrateStrategy(): FlywayMigrationStrategy {
        return FlywayMigrationStrategy { flyway ->
            flyway.clean()
            flyway.migrate()
        }
    }

}
